src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/stateful_transpose_sdpa_fusion.cpp

```diff
@@ -0,0 +1,156 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "stateful_transpose_sdpa_fusion.hpp"
+
+#include <cstdint>
+#include <limits>
+#include <openvino/core/rt_info.hpp>
+#include <openvino/opsets/opset1.hpp>
+#include <openvino/opsets/opset13.hpp>
+#include <openvino/opsets/opset6.hpp>
+#include <openvino/opsets/opset8.hpp>
+#include <openvino/pass/pattern/op/or.hpp>
+#include <openvino/pass/pattern/op/wrap_type.hpp>
+#include <transformations/utils/utils.hpp>
+
+#include "itt.hpp"
+#include "ov_ops/type_relaxed.hpp"
+#include "transformations/cpu_opset/common/op/sdpa.hpp"
+
+namespace ov {
+namespace intel_cpu {
+
+StatefulTransposeSDPAFusion::StatefulTransposeSDPAFusion() {
+    MATCHER_SCOPE(StatefulTransposeSDPAFusion);
+    using namespace ov::pass::pattern;
+
+    auto past_k = wrap_type<opset6::ReadValue>();
+    auto past_v = wrap_type<opset6::ReadValue>();
+    auto convert_past_k = wrap_type<opset1::Convert>({past_k});
+    auto convert_past_v = wrap_type<opset1::Convert>({past_v});
+    auto concat_input_k = std::make_shared<ov::pass::pattern::op::Or>(OutputVector{past_k, convert_past_k});
+    auto concat_input_v = std::make_shared<ov::pass::pattern::op::Or>(OutputVector{past_v, convert_past_v});
+    auto concat_k = wrap_type<opset6::Concat>({concat_input_k, any_input()});
+    auto concat_v = wrap_type<opset6::Concat>({concat_input_v, any_input()});
+
+    // multi-query branch
+    auto reshape_k = wrap_type<opset6::Reshape>({concat_k, any_input()});
+    auto reshape_v = wrap_type<opset6::Reshape>({concat_v, any_input()});
+    auto constant_k = wrap_type<opset6::Constant>();
+    auto constant_v = wrap_type<opset6::Constant>();
+    auto multiply_k = wrap_type<opset6::Multiply>({reshape_k, constant_k});
+    auto multiply_v = wrap_type<opset6::Multiply>({reshape_v, constant_v});
+    auto reshape1_k = wrap_type<opset6::Reshape>({multiply_k, any_input()});
+    auto reshape1_v = wrap_type<opset6::Reshape>({multiply_v, any_input()});
```

Comment:
There is no checks to make sure the values from constant_k & constant_v are all 1s.

Reply:
done, thanks!